Network/Socket/Options.hsc

```diff
@@ -39,10 +43,10 @@ import Network.Socket.Types
 --
 -- The existence of a constructor does not imply that the relevant option
 -- is supported on your system: see 'isSupportedSocketOption'
-data SocketOption = SockOpt {
-    sockOptLevel :: !CInt
-  , sockOptName  :: !CInt
-  } deriving (Eq, Show)
+data SocketOption = SockOpt
+    !CInt -- ^ Option Level
+    !CInt -- ^ Option Name
+  deriving (Eq)
```

Comment:
It turns out that the version of `doctest` with GHC prior to 8.6 fails to correctly parse the haddock comments in the above `SocketOption` declaration.  Removing the `^` marks makes it go.  Given that the data type is in any case opaque to users, the haddock-style comments are not essential, so best to just drop the `^` marks and let the code build with older GHC 8.x versions.